requestor/ext/requestor-gwtjackson/src/main/java/io/reinert/requestor/ext/gwtjackson/codegen/TypeInfo.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2015 Danilo Reinert
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.reinert.requestor.ext.gwtjackson.codegen;
+
+import java.util.ArrayList;
+
+import com.squareup.javapoet.ClassName;
+
+/**
+ * Info about some type that will be handled for code generation.
+ *
+ * @author Danilo Reinert
+ */
+public class TypeInfo {
+
+    private final Package pack;
+    private final String simpleName;
+    private final String qualifiedName;
+    private final ClassName className;
+    private final ArrayList<String> names = new ArrayList<String>();
+    private boolean innerType;
+
+    public TypeInfo(String qualifiedName) {
+        final String[] parts = qualifiedName.split("\\.");
```

Comment:
Agree. This is actually a class that will be part of another module, a private one for this project to help with code generation. In this case it's a required logic to separe parts of the given qualified name. Maybe I could refactor this constructor as a factory method (but I would have to do it with all others too, to maintain same pattern)